### PR TITLE
ACCUMULO-4178: Updated the sender thread to include the table ids.

### PR DESCRIPTION
**Pull request by ivakegg**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java [hunk 1]
```diff
@@ -831,11 +814,16 @@ public void send(TabletServerMutations<Mutation> tsm) throws AccumuloServerExcep
...
+          Set<Text> tableIds = new TreeSet<Text>();
+          for (Map.Entry<KeyExtent,List<Mutation>> entry : mutationBatch.entrySet()) {
+            count += entry.getValue().size();
+            tableIds.add(entry.getKey().getTableId());
           }
-          String msg = "sending " + String.format("%,d", count) + " mutations to " + String.format("%,d", mutationBatch.size()) + " tablets at " + location;
-          Thread.currentThread().setName(msg);
+          StringBuilder msg = new StringBuilder();
+          msg.append("sending ").append(String.format("%,d", count)).append(" mutations to ").append(String.format("%,d", mutationBatch.size()))
+              .append(" tablets for ").append(String.format("%,d", tableIds.size())).append(" tables ").append(tableIds).append(" to ").append(location);
```
Comment (dlmarion) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: nitpick - suggest adding surrounding tableIds with [] or (). Or, remove tableIds.size() from the output if you are going to list the tables.

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java [hunk 2]
```diff
@@ -831,11 +814,16 @@ public void send(TabletServerMutations<Mutation> tsm) throws AccumuloServerExcep
...
-            count += list.size();
+
+          Set<Text> tableIds = new TreeSet<Text>();
+          for (Map.Entry<KeyExtent,List<Mutation>> entry : mutationBatch.entrySet()) {
+            count += entry.getValue().size();
+            tableIds.add(entry.getKey().getTableId());
           }
-          String msg = "sending " + String.format("%,d", count) + " mutations to " + String.format("%,d", mutationBatch.size()) + " tablets at " + location;
-          Thread.currentThread().setName(msg);
+          StringBuilder msg = new StringBuilder();
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: When java generates code for string concatenation, it uses StringBuilder.   Is there any value in this case for using StringBuilder over concatenation?

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java [hunk 3]
```diff
@@ -16,37 +16,7 @@
  */
 package org.apache.accumulo.core.client.impl;
 
-import java.io.IOException;
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: What did you use for import manipulation?  I don't think the Accumulo project has declared how imports have been organized.  However I think we do try to organize imports using Eclipse's default settings inorder to cut down on noise generated by different IDEs organizing imports in different ways.  Can we just drop this portion of the patch?  Did the code add any imports?

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java (hunk 1 above)
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: Usually the toString method for collections will enclose the output with something like [].  I suspect TreeSet does this.

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java (hunk 2 above)
Comment (joshelser) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: I'm not aware of any positive effects of using a StringBuilder here. Like you said, Java is going to do that under the hood anyways.

**Comment by joshelser** [comment]
Keith and Dave summed up the comments that I had to add already :)

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java (hunk 1 above)
Comment (ivakegg) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: TreeSet already adds [].  Perhaps I should not depend on a toString method however and do the work myself to ensure it is added correctly.

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java (hunk 2 above)
Comment (ivakegg) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: fair enough, but is there any downside to using it here?  Especially if I am going to add the table ids myself instead of depending on the TreeSet implementation of toString()....

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java (hunk 3 above)
Comment (ivakegg) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: Sure, I will undo the reorganizing.  Sorry about that.

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java (hunk 1 above)
Comment (ivakegg) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: Also I agree including the number of table ids is redundant.  I will remove that part....

File: core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java (hunk 2 above)
Comment (joshelser) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: If you're going to build up `msg` inside of the for loop, I think it makes sense to use the StringBuilder. Just in the current state, it didn't look like there was any benefit to the use of a StringBuilder (so it just came across as an unnecessary change).

Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/client/impl/TabletServerBatchWriter.java]: > fair enough, but is there any downside to using it here?

In my opinion it makes the code a little harder for humans to parse.

**Comment by keith-turner** [comment]
Could put the table ids at the end of the message.

For example the following

```
sending 20 mutations to 5 tablets for [5a] to 192.168.1.103
sending 200 mutations to 10 tablets for [ab,cd,ef,gh,ij,k1,mn,rst,xyz] to 192.168.1.103
```

vs

```
sending 20 mutations to 5 tablets at 192.168.1.103 tids: [5a]
sending 200 mutations to 10 tablets at 192.168.1.103 tids: [ab,cd,ef,gh,ij,k1,mn,rst,xyz]
```

**Comment by ivakegg** [comment]
> Could put the table ids at the end of the message.

I do not have a real opinion either way, but if anybody has built tools to parse this message/title then I expect putting them at the end would probably be better.  I will make that change.

**Comment by ivakegg** [comment]
I think I have addresses all concerns here.  Any votes for merging this in?

**Comment by joshelser** [comment]
LGTM :+1:

I'll work on merging this in. Thanks, @ivakegg!

**Comment by keith-turner** [comment]
+1
